File: AI_interviewer/backstage/app/schemas/Interview_record.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class InterviewRecordBase(BaseModel):
    position_id: int
    interviewer_id: int

class InterviewRecordCreate(InterviewRecordBase):
    pass

class InterviewRecord(InterviewRecordBase):
    id: int
    user_id: int
    time: datetime
    
    # 为了方便前端展示，可以包含关联的名称信息
    position_name: Optional[str] = None
    interviewer_name: Optional[str] = None

    class Config:
        orm_mode = True

File: AI_interviewer/client/src/api/Interview_record.ts
import request from '../utils/request'

export interface InterviewRecord {
  id: number
  user_id: number
  position_id: number
  interviewer_id: number
  time: string
  position_name?: string
  interviewer_name?: string
}

export interface CreateRecordParams {
  position_id: number
  interviewer_id: number
}

export function createRecord(data: CreateRecordParams) {
  return request<any, InterviewRecord>({
    url: '/api/interviewee/create_record',
    method: 'post',
    data
  })
}

export function getRecords() {
  return request<any, InterviewRecord[]>({
    url: '/api/interviewee/get_records',
    method: 'get'
  })
}
